### PR TITLE
[1.9][FLINK-14104][build] Add opt-in jackson 2.10.1 profile

### PR DESCRIPTION
**Pull request by zentol**
This PR adds an opt-in profile that bumps jackson dependencies to 2.10.1. This version contains fixes for a large number of security fixes.
This upgrade is an opt-in feature as we usually don't bump dependencies for bugfix releases, and this is done shortly before the next release, giving us little time to let it mature.

green cron job that I ran yesterday: https://travis-ci.com/flink-ci/flink/builds/136505913

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -378,6 +378,24 @@ under the License.
 				<version>1.1.3</version>
 			</dependency>
 
+			<dependency>
```
Comment (hequn8128) [inline pom.xml]: Looks good!
Only one minor comment: add a profile for this?

Comment (zentol) [reply pom.xml]: very nice catch; this should of course be in a profile.

**Comment by zentol** [comment]
@hequn8128 I've updated the PR.

**Comment by hequn8128** [comment]
@zentol Thanks a lot for the update. Looks very good! +1

**Comment by zentol** [comment]
@flinkbot run travis

**Pull request merged**